### DISCUSSION TITLE
Erro Update/Create Database Migrations (Fundamentos do ASP.NET 6)

### DISCUSSION THREAD

**Discussion by LuizHenriqueLajus** [balta-io/forum#116] [Tira Dúvidas] [answered]
Boa tarde professor,

Estou tendo esse erro quando você fala pra gente Deletar ou dar Update no SqlServer. 
Segue o erro:

Blog\Controllers\AccountController.cs(14,38): error CS0161: 'AccountController.Post(RegisterViewModel, BlogDataContext)': not all code paths return a value

Já tentei tirar o código e realizar o migration, deletar a pasta do migrations e criar um novo e eles continuam a dar o mesmo erro. Preciso de uma ajuda pois não estou conseguindo resolver e desta maneira não consigo progredir no curso.

Atenciosamente,

Luiz Henrique

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @LuizHenriqueLajus, como vai?

Tem o código do AccountController ai pra gente dar uma olhada?

**Reply by LuizHenriqueLajus** [reply]
Até o momento onde acontece a solicitação pra gente deletar o banco ou dar update (que foi as duas tentativas que tentei realizar).
O código está assim:

using Blog.Models;
using EFBlog.Data;
using EFBlog.Extensions;
using EFBlog.Services;
using EFBlog.ViewModels;
using Microsoft.AspNetCore.Mvc;

namespace EFBlog.Controllers;

[ApiController]
public class AccountController : ControllerBase
{
    [HttpPost("v1/accounts/")]
    public async Task<IActionResult> Post(
        [FromBody] RegisterViewModel model,
        [FromServices] BlogDataContext context)
    {
        if (!ModelState.IsValid)
            return BadRequest(new ResultViewModel<string>(ModelState.GetErrors()));

        var user = new User
        {
            Name = model.Name,
            Email = model.Email,
            Slug = model.Email.Replace("@", "-").Replace(".", "-")
        };


        return Ok();
    }

    [HttpPost("v1/login")]
    public IActionResult Login([FromServices]TokenService tokenService)
    {
        var token = tokenService.GenerateToken(null);

        return Ok(token);
    }
}

**Reply by LuizHenriqueLajus** [reply]
Decidi reiniciar o curso do zero hoje. Acredito que resolva, irei fazer com mais cuidado.

**Reply by andrebaltieri** [reply]
Bom dia, @LuizHenriqueLajus, show... as vezes passou algo batido... é normal!!
Se precisar de ajuda, estamos aqui!
